fe/fe-core/src/main/java/com/starrocks/external/iceberg/IcebergUtil.java

```diff
@@ -162,4 +182,82 @@ public static Map<PartitionField, Integer> getIdentityPartitions(PartitionSpec p
         }
         return columns.build();
     }
+
+    public static IcebergTable convertToSRTable(org.apache.iceberg.Table icebergTable,
+                                                String resourceName, String dbName, String tblName) throws DdlException {
+        Map<String, String> properties = new HashMap<>();
+        properties.put(IcebergTable.ICEBERG_CATALOG, "HIVE_CATALOG");
+        properties.put(IcebergTable.ICEBERG_DB, dbName);
+        properties.put(IcebergTable.ICEBERG_TABLE, tblName);
+        properties.put(IcebergTable.ICEBERG_METASTORE_URIS, resourceName);
+        properties.put(IcebergTable.ICEBERG_RESOURCE, resourceName);
+        Map<String, Types.NestedField> icebergColumns = icebergTable.schema().columns().stream()
+                .collect(Collectors.toMap(Types.NestedField::name, field -> field));
+        List<Column> fullSchema = Lists.newArrayList();
+        for (Map.Entry<String, Types.NestedField> entry : icebergColumns.entrySet()) {
+            Types.NestedField icebergColumn = entry.getValue();
+            Type srType = convertColumnType(icebergColumn.type());
+            Column column = new Column(icebergColumn.name(), srType, true);
+            fullSchema.add(column);
+        }
+
+        return new IcebergTable(connectorTableIdIdGenerator.getNextId().asInt(), icebergTable.name(),
+                fullSchema, properties);
+    }
+
+    private static Type convertColumnType(org.apache.iceberg.types.Type icebergType) {
+        if (icebergType == null || !icebergType.isPrimitiveType()) {
+            return Type.NULL;
+        }
+
+        if (icebergType.isListType()) {
+            return convertColumnType(icebergType.asListType().elementType());
+        }
+
+        PrimitiveType primitiveType;
+
+        switch (icebergType.typeId()) {
+            case BOOLEAN:
+                primitiveType = PrimitiveType.BOOLEAN;
+                break;
+            case INTEGER:
+                primitiveType = PrimitiveType.INT;
+                break;
+            case LONG:
+                primitiveType = PrimitiveType.BIGINT;
+                break;
+            case FLOAT:
+                primitiveType = PrimitiveType.FLOAT;
+                break;
+            case DOUBLE:
+                primitiveType = PrimitiveType.DOUBLE;
+                break;
+            case DATE:
+                primitiveType = PrimitiveType.DATE;
+                break;
+            case TIMESTAMP:
+                primitiveType = PrimitiveType.DATETIME;
+                break;
+            case STRING:
+            case UUID:
+                primitiveType = PrimitiveType.VARCHAR;
+                break;
+            case DECIMAL:
+                primitiveType = PrimitiveType.DECIMAL32;
+                break;
+            case TIME:
+            case FIXED:
+            case BINARY:
+            case STRUCT:
+            case LIST:
```

Comment:
Array is supported

Reply:
List type is implemented here.  https://github.com/mxdzs0612/starrocks/blob/fe/fe-core/src/main/java/com/starrocks/external/iceberg/IcebergUtil.java#L213